Migrate animations to go_router

miquelbeltran opened the pull request:
In this PR, I am replacing the named routes implementation in the `animations` sample for one using `go_router`.

The codelab_rebuild.yaml would need to be updated to add the go_router installation step.

cc. @domesticmouse @RedBrogdon 

## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I read the [Contributors Guide].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

<!-- Links -->
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

miquelbeltran commented on animations/lib/main.dart:
```diff
@@ -147,29 +149,38 @@ final miscDemos = [
+      routes: [
+        for (final demo in basicDemos)
+          GoRoute(
+            path: demo.route.substring(1),
```
Need to remove the starting `/` from the original routes.

If you think this is a bit too hacky, I can go ahead and modify all the route names inside their classes.

johnpryan replied:
I'd vote to update the `route` values instead.

johnpryan reviewed (approved):
LGTM overall

domesticmouse commented:
Needs code formatting

```console
Running "flutter pub get" in animations...                       2,615ms
Analyzing animations...
No issues found!
Changed lib/main.dart
Formatted 28 files (1 changed) in 0.59 seconds.
```

miquelbeltran commented:
I changed the paths to remove the initial `/` char (and renamed a couple of them that were on the wrong list)

Let me know if you prefer this version, otherwise I'll revert the commit

johnpryan reviewed (approved):
Looks good to me 👍

domesticmouse commented:
@miquelbeltran press the Squash and merge button when you are happy that this is ready to land.

miquelbeltran commented:
> press the Squash and merge button when you are happy that this is ready to land.

@domesticmouse I don't have permissions to do that :wink: but it is ready to land now

domesticmouse commented:
> > press the Squash and merge button when you are happy that this is ready to land.
> 
> 
> 
> @domesticmouse I don't have permissions to do that :wink: but it is ready to land now
> 
> 

Whups, my bad 😏
